tools/clang/include/clang/AST/OperationKinds.h

```diff
@@ -314,6 +314,8 @@ enum CastKind {
   CK_HLSLVectorToMatrixCast,
   CK_HLSLMatrixToVectorCast,
   CK_HLSLDerivedToBase,
+  CK_HLSLRowMajorToColMajor,
+  CK_HLSLColMajorToRowMajor,
```

Comment:
Can you explain why you need a cast kind for this instead of this just being an extension of the FlatConversion?

Reply:
It is clearer than reuse FlatConversion.
The data is changed for major change. For FlatConversion, the data is not changed, more like a bitcast.